src/main/java/org/spongepowered/api/text/title/TitleBuilder.java

```diff
@@ -24,91 +24,311 @@
+     *
+     * @return {@code true} if the title will clear
+     * @see Title#isClear()
      */
-    TitleBuilder fadeOut(int ticks);
+    public final boolean isClear() {
+        return this.clear;
+    }
 
     /**
-     * Remove the currently displayed title from the player's screen. This will
+     * Removes the currently displayed title from the player's screen. This will
      * keep the currently used display times and will only remove the title.
      *
      * @return This title builder
+     * @see Title#isClear()
+     */
+    public TitleBuilder clear() {
+        return clear(true);
+    }
+
+    /**
+     * Sets whether the the currently displayed title should be removed from the
+     * player's screen and will keep the currently used display times.
+     *
+     * @param clear Whether this title should clear
+     * @return This title builder
+     * @see Title#isClear()
+     */
+    public TitleBuilder clear(boolean clear) {
+        if (this.clear = clear) {
+            this.title = null; // No need to send title if we clear it after
+                               // that again
+        }
+        return this;
+    }
+
+    /**
+     * Returns whether this builder is currently configured to reset.
+     *
+     * @return {@code true} if the title will reset
+     * @see Title#isReset()
+     */
+    public final boolean isReset() {
+        return this.reset;
+    }
+
+    /**
+     * Removes the currently displayed title from the player's screen and set
+     * the configuration back to the default values.
+     *
+     * @return This title builder
+     * @see Title#isReset()
      */
-    TitleBuilder clear();
+    public TitleBuilder reset() {
+        return reset(true);
+    }
 
     /**
-     * Remove the currently displayed title from the player's screen and set the
-     * configuration back to the default values.
+     * Sets whether the currently displayed title should be removed from the
+     * player's screen and the configuration set back to the default values.
      *
+     * @param reset Whether this title should reset
      * @return This title builder
+     * @see Title#isReset()
      */
-    TitleBuilder reset();
+    public TitleBuilder reset(boolean reset) {
+        if (this.reset = reset) {
+            // No need for these if we reset it again after that
+            this.title = null;
+            this.subtitle = null;
+            this.fadeIn = null;
+            this.stay = null;
+            this.fadeOut = null;
+        }
+        return this;
+    }
 
     /**
      * Builds an immutable instance of the current configuration.
      *
      * @return An immutable {@link Title} with the currently configured settings
      */
-    Title build();
+    public Title build() {
+        // Special case if the builder is empty
+        if (this.title == null && this.subtitle == null && this.fadeIn == null && this.stay == null && this.fadeOut == null) {
+            if (this.clear) {
```

Comment:
this logic here could be explained with comments, e.g. distinction between cleared and reset titles